usr/src/cmd/zdb/zdb.c

```diff
@@ -2121,6 +2130,153 @@ dump_label_uberblocks(vdev_label_t *lbl, uint64_t ashift)
 	}
 }
 
+static int
+dump_specific_file(objset_t *os, uint64_t obj)
+{
+	int print_header = 0;
+
+	if (dump_opt['v'] > 5)
+		print_header = 1;
+	dump_object(os, obj, dump_opt['v'], &print_header);
+	return (0);
+}
+
+static int
+dump_specific_directory(objset_t *os, uint64_t obj, char *name)
```

Comment:
This takes a path to a file or directory, right?  Might make more sense to name this dump_specific_path?